packages/common/src/doctype.ts

```diff
@@ -18,56 +18,85 @@ export enum AnchorStatus {
     NOT_REQUESTED, PENDING, PROCESSING, ANCHORED, FAILED
 }
 
-/**
- * Describes anchor record
- */
-export interface AnchorRecord {
-    prev: CID; // should be CID type
-    proof: CID; // should be CID type
-    path: string;
+export interface RecordHeader {
+    controllers: Array<string>
+    schema?: string
+    tags?: Array<string>
+
+    unique?: string
+
+    [index: string]: any // allow support for future changes
+}
+
+export type GenesisHeader = RecordHeader
+
+export type GenesisRecord = {
+    header: GenesisHeader
+    data?: any
+    unique?: string
+}
+
+export interface UnsignedRecord {
+    id: CID
+    header?: RecordHeader
+    data: any
+    prev: CID
 }
 
-/**
- * Describes anchor proof
- */
 export interface AnchorProof {
-    chainId: string;
-    blockNumber: number;
-    blockTimestamp: number;
-    txHash: CID;
-    root: CID;
+    chainId: string,
+    blockNumber: number,
+    blockTimestamp: number,
+    txHash: CID,
+    root: CID,
+}
+
+export interface AnchorRecord {
+    id: CID,
+    prev: CID,
+    refs?: Array<CID>
+    proof: CID,
+    path: string,
 }
 
+import type { DagJWSResult, DagJWS } from 'dids'
+
+export type SignedRecordDTO = DagJWSResult
+
+export type SignedRecord = DagJWS
+
+export type CeramicRecord = GenesisRecord | UnsignedRecord | AnchorRecord | SignedRecordDTO | SignedRecord
+
 /**
  * Document metadata
  */
 export interface DocMetadata {
-    controllers: Array<string>;
-    schema?: string;
-    tags?: Array<string>;
+    controllers: Array<string>
+    schema?: string
+    tags?: Array<string>
 
-    [index: string]: any; // allow arbitrary properties
+    [index: string]: any // allow arbitrary properties
 }
 
 /**
  * Document params
  */
 export interface DocParams {
-    metadata?: DocMetadata;
+    metadata?: DocMetadata
     // deterministic is a tri-state. True means means always create the document deterministically,
     // false means always force the document to be unique, undefined means use the default behavior
-    deterministic?: boolean;
+    deterministic?: boolean
 
-    [index: string]: any; // allow arbitrary properties
+    [index: string]: any // allow arbitrary properties
```

Comment:
Do we want this? Having explicit props would make it easier to know what is supported here.

Reply:
I agree. I'll remove it

Reply:
actually, `DocParams` can have arbitrary properties depending on the doctype (`content` for tile, etc.)

Reply:
Maybe that could be handled dynamically then? Something like
```ts
DocParams<T> = {
  deterministic?: boolean
  metadata?: DocMetadata
  doctype: T
}
```
This way we could still have an explicit key and type for the doctype-specific params?

Reply:
@PaulLeCam hm, not sure if that will solve the problem. For example, `TileParams`:

```
interface TileParams extends DocParams {
    content?: Record<string, unknown>;
}
```

Reply:
OK I see, I think we could use method overrides then?
```ts
createDocument('caip10-link', Caip10LinkParams): ...
createDocument('tile', TileParams): ...
...
```

Reply:
The `createDocument` in the API should become something like:

```
createDocument<T extends Doctype, D extends DocParams>(doctype: string, params: D, opts?: DocOpts): Promise<T>;
```

Maybe I didn't get it 🤔  Do you have something else in mind?

Reply:
The issue with `<T extends Doctype, D extends DocParams>` is that `T = TileDoctype` and `D = Caip10LinkParams` would be valid type-wise even though it can create runtime errors, so I think that should be avoided.
With the code I suggested above, having `createDocument('tile', Caip10LinkParams)` would be a compilation error.